graph/depth_first_search[easier approach].cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ *
+ * \file
+ * \brief [Depth First Search Algorithm
+ * (Depth First Search)](https://en.wikipedia.org/wiki/Depth-first_search)
+ *
+ * \author [Khalid Abdullah](http://github.com/khalid586)
+ *
+ * \details
+ * Depth First Search also quoted as DFS is a Graph Traversal Algorithm.
+ * Time Complexity O(|V| + |E|) where V is number of vertices and E
+ * is number of edges in graph.
+ *
+ * Application of Depth First Search are
+ *
+ * 1. Finding connected components
+ * 2. Finding 2-(edge or vertex)-connected components.
+ * 3. Finding 3-(edge or vertex)-connected components.
+ * 4. Finding the bridges of a graph.
+ * 5. Generating words in order to plot the limit set of a group.
+ * 6. Finding strongly connected components.
+ *
+ * And there are many more...
+ *
+ * <h4>Working</h4>
+ * 1. Mark all vertices as unvisited first
+ * 2. start exploring from some starting vertex.
+ *
+ *      While exploring vertex we mark the vertex as visited
+ *      and start exploring the vertices connected to this
+ *      vertex in recursive way.
+ *
+ */
+
+#include <bits/stdc++.h>
+using namespace std;
+
+const int N = 1e7 + 1;
+int vis[N];  // array for marking the nodes as visited or unvisited
+std::vector<int> graph[N];  // graph given by the user
```

Comment:
Hi @khalid586 , instead of creating array for `vis` and vector for `graph` , we can create them in main function as per input and pass their reference in DFS function so that it will not use extra space if input it small